### ISSUE TITLE
[a11y]: Export Modal, Focus moves to parent page

### ISSUE THREAD

**Issue by devadula-nandan** [carbon-design-system/ibm-products#6021]
### Package

@carbon/ibm-products

### Browser

Firefox

### Operating System

Windows

### Package version

v2.49.0-rc.0

### Automated testing tool and ruleset

manual

### Assistive technology

JAWS

### Description

Reported by Raghu during QA Audit.

After activating the "Export" button, the file is exported but the focus moves behind the dialog on the parent page.

### Recommended fix
Once the Modal opens, the focus is expected to be set to the first interactive element within the Modal, here focus should not move out  /Navigate /Read any content of the parent page.- Activation /selection of any control should not cause any focus jumps.

### User Impact
After a user action, if the focus randomly jumps to another spot on the web page, it’s tedious to navigate back to the relevant element. This negatively affects the experience of keyboard-only users and those with cognitive disabilities.

### WCAG 2.1 Violation

2.4.3 Focus Order

### Reproduction/example

https://ibm-products.carbondesignsystem.com/?path=/story/ibm-products-components-export-exportmodal--standard&globals=viewport:basic

### Steps to reproduce

open the export modal and click on the export button, the focus shifts to the page

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/ibm-products/blob/main/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/ibm-products/issues) for duplicate problems

**Comment by devadula-nandan** [comment]
related [issue](https://github.com/carbon-design-system/ibm-products/issues/6016)
